pkg/core/xds/matched_policies.go

```diff
@@ -1,23 +1,68 @@
 package xds
 
 import (
+	"encoding"
 	"fmt"
 	"sort"
 
+	"google.golang.org/protobuf/proto"
+
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
 	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
 )
 
 type MatchingPolicyMap map[core_model.ResourceType][]core_model.Resource
 
+type GetTargetRef interface {
```

Comment:
It was difficult to come up with a name for this interface, so I named it like a method :) 
By `Matchable` I understand something that could be matched, for example, DPP is "Matchable" because it could be matched by policies. `Attachable` makes sense only for top-level targetRef, items in `to/from` lists are not "Attachable" in my understanding.

Reply:
Ok, is this interface even used?

We could inline it in interfaces below, and call `ResourceSpecWithTargetRef` a `Policy`, since Policy for us is something that has a spec and can be attached to things.

Reply:
Done